hypertrace-graphql-entity-schema/src/test/java/org/hypertrace/graphql/entity/joiner/DefaultEntityJoinerBuilderTest.java

```diff
@@ -67,14 +71,17 @@
   @Mock DataFetchingFieldSelectionSet mockSelectionSet;
   @Mock AttributeAssociation<FilterArgument> mockFilter;
   @Mock ResultSetRequest mockResultSetRequest;
-  @Mock EntityLabelRequestBuilder mockEntityLabelRequestBuilder;
+  EntityLabelRequestBuilder entityLabelRequestBuilder;
 
   Scheduler testScheduler = Schedulers.single();
 
   DefaultEntityJoinerBuilder entityJoinerBuilder;
 
   @BeforeEach
   void beforeEach() {
+    this.entityLabelRequestBuilder =
+        new DefaultEntityLabelRequestBuilder(attributeRequestBuilder, mockSelectionFinder);
```

Comment:
Do we need the real thing here? Could we mock this (and leave the class package private), instead of mocking the selection fields that will be used by the implementation?